tests/test_premium.py

```diff
@@ -0,0 +1,422 @@
+    mock_stripe.Price.search.return_value = []
+    mock_stripe.Price.create.return_value = MagicMock(id="price_123")
+
+    with app.app_context():
+        tier = Tier.query.filter_by(name="Business").first()
+        assert tier is not None
+        if not tier:
+            return
+
+        update_price(tier)
+
+        assert mock_stripe.Price.create.called
+        assert mock_stripe.Product.modify.called
+
+
+def test_create_customer(app: Flask, mock_stripe: MagicMock) -> None:
+    mock_stripe.Customer.create.return_value = MagicMock(id="cus_123")
+
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        db.session.add(user)
+        db.session.commit()
+
+        create_customer(user)
+
+        assert mock_stripe.Customer.create.called
+        assert user.stripe_customer_id is not None
+
+        # Check that the customer ID was saved to the database
+        db.session.refresh(user)
+        assert user.stripe_customer_id is not None
+
+
+def test_create_subscription(app: Flask, mock_stripe: MagicMock) -> None:
+    mock_stripe.Customer.create.return_value = MagicMock(id="cus_123")
+    mock_stripe.Subscription.create.return_value = MagicMock(id="sub_123")
+
+    with app.app_context():
+        tier = Tier.query.filter_by(name="Business").first()
+        assert tier is not None
+
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        db.session.add(user)
+        db.session.commit()
+
+        create_subscription(user, tier)
+
+        assert mock_stripe.Subscription.create.called
+        assert user.stripe_subscription_id is not None
+
+        # Check that the subscription ID was saved to the database
+        db.session.refresh(user)
+        assert user.stripe_subscription_id is not None
+
+
+def test_get_latest_invoice_payment_intent_client_secret(mock_stripe: MagicMock) -> None:
+    mock_stripe.Invoice.retrieve.return_value = MagicMock(payment_intent="pi_123")
+    mock_stripe.PaymentIntent.retrieve.return_value = MagicMock(client_secret="secret_123")  # noqa: S106
+
+    secret = get_latest_invoice_payment_intent_client_secret(MagicMock(latest_invoice="inv_123"))
+
+    assert secret == "secret_123"
+
+
+def test_get_subscription(app: Flask, mock_stripe: MagicMock) -> None:
+    mock_stripe.Subscription.retrieve.return_value = MagicMock(id="sub_123")
+
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
+        user.stripe_subscription_id = "sub_123"
+        db.session.add(user)
+        db.session.commit()
+
+        subscription = get_subscription(user)
+
+        assert subscription is not None
+        assert mock_stripe.Subscription.retrieve.called
+
+
+def test_handle_subscription_created(app: Flask) -> None:
+    with app.app_context():
+        user = User(email="test@example.com", password="password")  # noqa: S106
```

Comment:
`user` fixture

Reply:
Fixed in 180d5b3514b575c43e6b99320d8471a6f341796f